File: color_extractor.js
const fs = require('fs');
const path = require('path');

// Using a simple approach - analyze image with sharp if available, or use pixel-by-pixel if not
try {
  const sharp = require('sharp');
  
  async function analyzeImage() {
    const imagePath = path.join(__dirname, 'src', 'services', 'design', 'Home.png');
    
    const image = sharp(imagePath);
    const metadata = await image.metadata();
    console.log(`Image dimensions: ${metadata.width}x${metadata.height}\n`);
    
    // Extract raw pixel data
    const buffer = await image.raw().toBuffer({ resolveWithObject: true });
    const pixelData = buffer.data;
    const channels = buffer.info.channels;
    
    console.log("=== COLOR ANALYSIS FROM HOME.PNG ===\n");
    
    // Helper to get average color from a region
    const getRegionColor = (x1, x2, y1, y2, label) => {
      let r = 0, g = 0, b = 0;
      let count = 0;
      
      for (let x = x1; x < x2; x += 3) {
        for (let y = y1; y < y2; y += 3) {
          const idx = (y * metadata.width + x) * channels;
          if (idx + 2 < pixelData.length) {
            r += pixelData[idx];
            g += pixelData[idx + 1];
            b += pixelData[idx + 2];
            count++;
          }
        }
      }
      
      r = Math.round(r / count);
      g = Math.round(g / count);
      b = Math.round(b / count);
      
      const hex = `#${r.toString(16).padStart(2, '0')}${g.toString(16).padStart(2, '0')}${b.toString(16).padStart(2, '0')}`.toUpperCase();
      console.log(`${label}: RGB(${r}, ${g}, ${b}) = ${hex}`);
      return hex;
    };
    
    // Sidebar (left ~170px)
    console.log("SIDEBAR:");
    getRegionColor(10, 160, 20, 150, "  Background (top)");
    getRegionColor(10, 160, 150, 400, "  Background (middle)");
    
    console.log("\nMAIN BACKGROUND:");
    getRegionColor(200, 900, 50, 150, "  Main content area");
    
    console.log("\nCARDS:");
    getRegionColor(205, 350, 125, 195, "  Stat card background");
    
    console.log("\nTEXT:");
    getRegionColor(215, 330, 138, 148, "  Card label text");
    getRegionColor(215, 330, 155, 180, "  Card value text");
  }
  
  analyzeImage().catch(err => {
    console.error('Sharp error:', err.message);
    console.log('\nSharp not available, using manual color analysis...');
  });
} catch (err) {
  console.error('Error loading modules:', err.message);
  console.log('Manual color extraction not available in this environment.');
}

File: color_extractor.py
from PIL import Image
import statistics

img = Image.open(r'src\services\design\Home.png')
pixels = img.load()
width, height = img.size

print(f"Image dimensions: {width}x{height}")

# Sample specific regions for color extraction
def sample_region(x_start, x_end, y_start, y_end, label):
    colors = []
    for x in range(x_start, min(x_end, width), 5):
        for y in range(y_start, min(y_end, height), 5):
            colors.append(pixels[x, y][:3])  # RGB only
    
    if colors:
        # Get median color values for more accurate sampling
        r_vals = [c[0] for c in colors]
        g_vals = [c[1] for c in colors]
        b_vals = [c[2] for c in colors]
        
        avg_r = int(statistics.median(r_vals))
        avg_g = int(statistics.median(g_vals))
        avg_b = int(statistics.median(b_vals))
        
        hex_color = f'#{avg_r:02x}{avg_g:02x}{avg_b:02x}'
        print(f"{label}: RGB({avg_r}, {avg_g}, {avg_b}) = {hex_color}")

print("\n=== COLOR ANALYSIS FROM HOME.PNG ===\n")

# Sidebar (left area - approximately 0-170 pixels wide)
print("SIDEBAR:")
sample_region(10, 160, 20, 150, "  Background (top area)")
sample_region(10, 160, 150, 300, "  Background (middle area)")
sample_region(10, 160, 600, 700, "  Background (bottom area)")

print("\nMAIN PAGE BACKGROUND:")
sample_region(200, 900, 50, 150, "  Main content area")

print("\nCARDS (Stat cards - top row):")
sample_region(205, 350, 125, 195, "  Card background")
sample_region(215, 330, 140, 162, "  Card value text")

print("\nCHART CARDS:")
sample_region(205, 650, 320, 400, "  Chart container background")

print("\nBORDERS:")
sample_region(203, 207, 120, 200, "  Card border (left edge)")

print("\nTEXT COLORS:")
sample_region(215, 330, 138, 148, "  Card label text (small)")
sample_region(215, 330, 160, 175, "  Card value text (large)")

print("\nBASE ELEMENTS:")
sample_region(800, 900, 10, 30, "  Notification bell area")
